tests/integration_tests.rs

```diff
@@ -185,3 +188,147 @@ async fn test_lndk_send_invoice_request() {
         }
     }
 }
+
+#[tokio::test(flavor = "multi_thread")]
+// Here we test that when the offer creator responds with an invoice, lndk is able to receive it.
+async fn test_lndk_wait_for_invoice() {
+    let test_name = "lndk_wait_for_invoice";
+    let (bitcoind, mut lnd, ldk1, ldk2, _lndk_dir) =
+        common::setup_test_infrastructure(test_name).await;
+
+    // Here we'll produce a little network of channels:
+    //
+    // ldk1 <- ldk2 <- lnd
+    //
+    // ldk1 will be the offer creator, which will build a blinded route from ldk2 to ldk1.
+    let (pubkey, addr) = ldk1.get_node_info();
+    let (pubkey_2, addr_2) = ldk2.get_node_info();
+
+    ldk1.connect_to_peer(pubkey_2, addr_2).await.unwrap();
+    lnd.connect_to_peer(pubkey_2, addr_2).await;
+
+    let lnd_info = lnd.get_info().await;
+    let lnd_pubkey = PublicKey::from_str(&lnd_info.identity_pubkey).unwrap();
+
+    // ldk1 won't send back the invoice unless we have channels in our network.
+    let ldk2_fund_addr = ldk2.bitcoind_client.get_new_address().await;
+    let lnd_fund_addr = lnd.new_address().await.address;
+
+    // We need to convert funding addresses to the form that the bitcoincore_rpc library recognizes.
+    let ldk2_addr_string = ldk2_fund_addr.to_string();
+    let ldk2_addr = bitcoincore_rpc::bitcoin::Address::from_str(&ldk2_addr_string)
+        .unwrap()
+        .require_network(RpcNetwork::Regtest)
+        .unwrap();
+    let lnd_addr = bitcoincore_rpc::bitcoin::Address::from_str(&lnd_fund_addr)
+        .unwrap()
+        .require_network(RpcNetwork::Regtest)
+        .unwrap();
+    let lnd_network_addr = lnd
+        .address
+        .replace("localhost", "127.0.0.1")
+        .replace("https://", "");
+
+    // Fund both of these nodes so we can open channels.
+    bitcoind
+        .node
+        .client
+        .generate_to_address(1, &ldk2_addr)
+        .unwrap();
+    bitcoind
+        .node
+        .client
+        .generate_to_address(1, &lnd_addr)
+        .unwrap();
+
+    // Make sure the above funds are confirmed.
+    bitcoind
+        .node
+        .client
+        .generate_to_address(100, &ldk2_addr)
+        .unwrap();
+
+    ldk2.open_channel(pubkey, addr, 200000, 0, true)
+        .await
+        .unwrap();
+    ldk2.open_channel(
+        lnd_pubkey,
+        SocketAddr::from_str(&lnd_network_addr).unwrap(),
+        200000,
+        100000,
+        false,
+    )
+    .await
+    .unwrap();
+
+    bitcoind
+        .node
+        .client
+        .generate_to_address(100, &ldk2_addr)
+        .unwrap();
+
+    // TODO: Restarting the lnd node here isn't ideal. But for some reason after we call LDK's open_channel
+    // API call, LND blocks and is unable to receive any API requests. This might be a bug in ldk-sample or
+    // our ldk-sample implementation.
+    lnd.handle.kill().expect("lnd couldn't be killed");
```

Comment:
Hmmm FYI I found a fix for this -- it seems like the problem has something to do with mining a bunch of blocks while lnd is trying to initiate a channel open. I'll push up a fix for this soon.

Reply:
Ah that makes sense - some RPC calls will fail if we're not synced to chain. Easiest way will prob be to poll `GetInfo` to make sure that we're synced?

Reply:
Good call, I'll give that a try. Right now I just have some sleeps scattered throughout as a POC, which is definitely not ideal!